src/libYARP_manager/src/manager.cpp

```diff
@@ -370,6 +391,22 @@ Broker* Manager::createBroker(Module* module)
     return (new ScriptYarprunBroker(module->getBroker()));
 }
 
+bool Manager::removeBroker(Executable* exe)
+{
+    if (exe == nullptr)
+    {
+        return false;
+    }
+    else if(exe->state() == RUNNING)
+    {
+        exe->stop();
+        exe->stopWatchDog();
+    }
+
+    exe->removeBroker(); //TODO possible race condition in case watchdog enabled.
```

Comment:
The race condition cannot be handled within `exec->removeBroker()` function?

Reply:
For this point I need your help because probably you know better than me how the watchdog works and where it is used. I didn't experienced this race condition neither running big applications like `iol` and I don't know how this broker-switch can give problems with the watchdog, since I call `exe->stopWatchDog();` before removing the broker. Any hint could be helpful 😉